DIPs/DIP1xxx-JS.md

```diff
@@ -0,0 +1,121 @@
+and should to describe what steps can be taken by the user to upgrade their codebase.
+
+In order to facilitate on schedule deprecations, a comment of the format
+`@@@DEPRECATED_[version]@@@` should be added to the top of the code to be removed/disabled.
+This comment allows code to be easily searched before every release to
+catch all planned deprecations.
+
+### Public Functions, Types, and Modules
+
+All removals (or changes that make the code `private`/`package`) of public functions,
+types, and modules must be accompanied with a deprecation period.
+
+The symbol(s) must be marked using the `deprecated` keyword with a message containing
+the planned removal period. A pointer to more information should also be added. E.g.
+"See the 2.080 changelog for more details" or "See the function documentation for more
+details". The documentation of the symbol(s) must be updated noting the
+deprecation and removal plan. The documentation should contain information to help
+the users using the symbol(s) transition their code away from the symbol(s).
+
+If the deprecation is occuring because the symbol(s) are being replaced by new
+symbols, both the old and the new symbol(s) should be availible un-deprecated
+in at least one release to allow users to build their code without issue on
+both the `stable` and `master` branches.
+
+On the fifth release in the deprecation period, the documentation for the symbol
+should be removed while keeping the code itself public until removal.
+
+If there is no equivalent for the functionality of the removed symbol in the
+standard library or the runtime, the code should be moved to
+[undeaD](https://github.com/dlang/undeaD) to allow users to keep their current
+code if refactoring is not possible.
+
+### Language Features
+
+Unless the removed language feature is very unsafe or causes damage to real
+world systems, all changes or removals must be accompanied with a deprecation
+period. "Language features" includes bugs in the current behavior that existing
+user code depends on, e.g. [Issue 10378](https://issues.dlang.org/show_bug.cgi?id=10378).
+Fixing such issues should include a deprecation period for the current behavior,
+and an introduction of the new behavior as the default only at the end of the
+period.
+
+Deprecations to language features must also update the [language deprecations
+page](https://dlang.org/deprecate.html) on dlang.org simultaneously. The deprecation
+message given by the compiler should contain the planned removal period and/or a
+pointer to more information pertaining to the deprecation.
+
+Warnings must NOT be used in the deprecation process. Warnings are set as errors
+in many build systems (including DUB), and would therefore prematurely break many
+user's code. The exception is when the deprecation is for a change which turns 
+something into a warning. In this case the code which would trigger the warning must
```

Comment:
"which turns something into a warning"

I think some clarification is needed here. I had to go back and read it twice. Perhaps "which causes warnings to be emitted where none were before."